### PR TITLE
Extract the models from the script, liven it up a bit.

### PR DESCRIPTION
**Pull request by perrygoy**
This PR splits up the original Animal Fight Club script into smaller, more focused files. This step will hopefully set us up for success as we add more features to Animal Fight Club.

Firstly, the models now live in `/models/*.py`. I added a base `Fighter` class which `Animal` inherits from, and a `Referee` class which runs the fights and does some fun announcements.

Secondly, i made it so any number of fighters can enter the fight, and made our roster randomized. Here's an example run:

```
We've got a great fight for you today folks, just look at these 9 fighters...
In the left-middle corner, still wearing their bathrobe, UNICORN!
And in the right corner, with a face only their mother could love, BASILISK!
And in the right corner, brushing their teeth with steel wool, WOLVERINE!
And in the kitty corner, building a card tower, KOMODO DRAGON!
And in the left-middle corner, with rage in their eyes, DUCK!
And in the top-left corner, texting their buddies back home, MOUSE!
And in the right-middle corner, all the way from Saskatoon, STUART!
And in the far corner, with rage in their eyes, NEWT!
And in the right corner, building a card tower, ELEPHANT!
Let's get READY TO RUMBLE!!!

All right everyone, i want a good clean fight. First to 100!

newt wins!
newt: ROAR!!!
```

### CODE REVIEW

File: models/referee.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    "still wearing their bathrobe",
+    "with a face only their mother could love",
+    "trembling like a newborn deer",
+    "looking like they don't know where they are",
+    "texting their buddies back home",
+    "building a card tower",
+    "brushing their teeth with steel wool",
+    "with rage in their eyes",
+    "like they've got somethin' to prove",
+]
```
Comment (Arrex3) [inline models/referee.py:37]: I would prefer to have these in their own files

File: models/referee.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        print("Let's get READY TO RUMBLE!!!\n")
+
+    def _get_pairs(self) -> List[Tuple[Fighter, Fighter]]:
+        """Pair the combatants up, probably for a round."""
+        pairings = list(zip(self.combatant_list, self.combatant_list[1:]))
+        pairings.append((self.combatant_list[-1], self.combatant_list[0]))
+
+        return pairings
+
+    def fight(self, to: int = 100) -> Fighter:
```
Comment (Arrex3) [inline models/referee.py:74]: "fight" should be "ref_round"

File: animal_fight_club.py [hunk 3]
```diff
@@ -1,38 +1,47 @@
...
+    "hoooah",
+    "huzzah",
+    "squeak",
+    "plap",
+    "groan",
+    "zzz",
+    "keekee",
+    "caw",
+    "honk",
+]
```
Comment (Arrex3) [inline animal_fight_club.py:39]: These should go in the same data file

**Review by Arrex3 — CHANGES_REQUESTED** [review]

File: models/referee.py (hunk 2 above)
Comment (perrygoy) [reply models/referee.py]: I changed it to `run_tournament` after looking up some definitions and stuff. How's that sound?

File: animal_fight_club.py (hunk 3 above)
Comment (perrygoy) [reply animal_fight_club.py]: ✅

**Review by Arrex3 — APPROVED** [review]

**Pull request merged**